tests/python_package_test/test_engine.py

```diff
@@ -48,2591 +48,2655 @@ def categorize(continuous_x):
+    metrics_combination_train_regression([lgb_valid1, lgb_valid2], ['l1', 'l2'], iter_min_l1, True)
+    metrics_combination_train_regression([lgb_valid2, lgb_valid1], ['l1', 'l2'], iter_min_l1, True)
+
+    # test for lgb.cv
+    metrics_combination_cv_regression(None, iter_cv_l2, True, False)
+    metrics_combination_cv_regression('l2', iter_cv_l2, True, False)
+    metrics_combination_cv_regression('l1', iter_cv_l1, True, False)
+    metrics_combination_cv_regression(['l2', 'l1'], iter_cv_l2, True, False)
+    metrics_combination_cv_regression(['l1', 'l2'], iter_cv_l1, True, False)
+    metrics_combination_cv_regression(['l2', 'l1'], iter_cv_min, False, False)
+    metrics_combination_cv_regression(['l1', 'l2'], iter_cv_min, False, False)
+    metrics_combination_cv_regression(None, iter_cv_l2, True, True)
+    metrics_combination_cv_regression('l2', iter_cv_l2, True, True)
+    metrics_combination_cv_regression('l1', iter_cv_l1, True, True)
+    metrics_combination_cv_regression(['l2', 'l1'], iter_cv_l2, True, True)
+    metrics_combination_cv_regression(['l1', 'l2'], iter_cv_l1, True, True)
+    metrics_combination_cv_regression(['l2', 'l1'], iter_cv_min, False, True)
+    metrics_combination_cv_regression(['l1', 'l2'], iter_cv_min, False, True)
+
+    # test feval for lgb.cv
+    metrics_combination_cv_regression('None', 1, False, False,
+                                      feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
+                                                                       constant_metric(preds, train_data)])
+    metrics_combination_cv_regression('None', 25, True, False,
+                                      feval=lambda preds, train_data: [decreasing_metric(preds, train_data),
+                                                                       constant_metric(preds, train_data)])
+    metrics_combination_cv_regression('None', 1, True, False,
+                                      feval=lambda preds, train_data: [constant_metric(preds, train_data),
+                                                                       decreasing_metric(preds, train_data)])
+
+
+def test_node_level_subcol():
+    X, y = load_breast_cancer(return_X_y=True)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'binary',
+        'metric': 'binary_logloss',
+        'feature_fraction_bynode': 0.8,
+        'feature_fraction': 1.0,
+        'verbose': -1
+    }
+    lgb_train = lgb.Dataset(X_train, y_train)
+    lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train)
+    evals_result = {}
+    gbm = lgb.train(params, lgb_train,
+                    num_boost_round=25,
+                    valid_sets=lgb_eval,
+                    verbose_eval=False,
+                    evals_result=evals_result)
+    ret = log_loss(y_test, gbm.predict(X_test))
+    assert ret < 4
```

Comment:
![image](https://user-images.githubusercontent.com/25141164/105410302-c81ef100-5c42-11eb-9014-33c4eea0cc88.png)

```suggestion
    assert ret <  0.14
```